### DISCUSSION TITLE
How to convert yyyy-mm-ddT00:00:00.000Z format to yyyy-mm-dd format?

### DISCUSSION THREAD

**Discussion by positivedeveloper36** [adonisjs/core#3849] [Help]
My query like this :
```
    const query = Period.query()
      .select('id', 'date_from')
    return query
```
If the query run and I check in the postman, the result of date_from is yyyy-mm-ddT00:00:00.000Z 

I search reference and I find to add this in the model like this ::
```
 @column.date()
  public dateFrom: DateTime | string
```
I check call from the postman and it works, the result : yyyy-mm-dd

I need to change result of my query like this ::

```

    const query = Period.query()
      .select('id', 'date_from')
    

    const result = {
        data: (await query).flatMap(
          (period) => period.Files.map((file) => ({
            ...
            date_from: period.dateFrom,
            ...
          }))
        )
      };
     return result
```
When I change the script like that and check the result, the format back change to yyyy-mm-ddT00:00:00.000Z 

I try add moment js like this : date_from: moment(period.dateFrom).format('YYYY-MM-DD'), it success to  yyyy-mm-dd format. But the date is the current date not the date in the table of database

How can I solve this problem?

Note ::

field in the migration like this : 
```
table.date('date_from').notNullable()
```

I use serialize like below. But it does not works
```
  @column.dateTime({
    autoCreate: true,
    serialize: (value: DateTime | null) => {
      return value ? value.setZone('utc').toISO() : value
    },
  })
  public dateFrom: DateTime

```

**Comment by Barbapapazes** [comment]
Hello,

Have you try to use luxonjs which is the package which provide the DateTime type? Using it with the format method can solve you're issue ([and using the serializer](https://docs.adonisjs.com/guides/models/serializing-models#mutatingtransforming-values))

https://moment.github.io/luxon/#/

**Reply by positivedeveloper36** [reply]
@Barbapapazes It seems that luxon is automatically installed when we install adonis for the first time. So no need to install it again. I try use serializer like this : ```  @column.dateTime({
    autoCreate: true,
    serialize: (value: DateTime | null) => {
      return value ? value.setZone('utc').toISO() : value
    },
  })
  public dateFrom: DateTime```. But it does not works. I updated my question use serialize

**Reply by positivedeveloper36** [reply]
@Barbapapazes I try this : `@column.date({
    serialize: (value) => value.toFormat('yyyy-LL-dd'),
})
public date: dateFrom`. If I return query it works. But if I return result, it does not work. Seems it does not work in flatMap function
